test/integration/kernel.spec.ts

```diff
@@ -7182,6 +7182,93 @@ describe('Kernel', () => {
 				});
 		});
 
+		it('should filter contracts by the options.mask schema if set', async () => {
+			const scope = uuid();
+			const schema: JsonSchema = {
+				required: ['data'],
+				properties: {
+					data: {
+						required: ['scope'],
+						properties: {
+							scope: {
+								const: scope,
+							},
+						},
+					},
+				},
+			};
+
+			const contract1 = await ctx.kernel.insertContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					type: 'card@1.0.0',
+					data: {
+						scope,
+						status: 'open',
+					},
+				},
+			);
+
+			const contract2 = await ctx.kernel.insertContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					type: 'card@1.0.0',
+					data: {
+						scope,
+						status: 'closed',
+					},
+				},
+			);
+
+			const stream = await ctx.kernel.stream(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				schema,
+			);
+
+			const queryWithoutMaskId = uuid();
+			const queryWithMaskId = uuid();
+
+			stream.on('dataset', async (payload) => {
+				if (payload.id === queryWithoutMaskId) {
+					expect(payload.cards).toEqual([contract1, contract2]);
+
+					const mask: JsonSchema = {
+						type: 'object',
+						properties: {
+							data: {
+								properties: {
+									status: {
+										const: 'open',
+									},
+								},
+							},
+						},
+					};
+
+					stream.emit('query', {
+						id: queryWithMaskId,
+						schema,
+						options: {
+							mask,
+						},
+					});
+				} else if (payload.id === queryWithMaskId) {
+					expect(payload.cards).toEqual([contract1]);
+					stream.close();
+				}
+			});
+
+			stream.emit('query', {
```

Comment:
Just need to add `sortBy` here since the `toEqual` call in the `dataset` handler depends on the order.